api/core/v1alpha1/oasf-validator/skills.go

```diff
@@ -0,0 +1,166 @@
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Create a new HTTP request with the context
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, OASFEndpoint, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Perform the HTTP request
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to fetch skills from OASF: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var rawSkills []map[string]any
+	if err := json.Unmarshal(body, &rawSkills); err != nil {
+		return fmt.Errorf("failed to unmarshal skills: %w", err)
+	}
+
+	for _, rawSkill := range rawSkills {
+		uid, ok := rawSkill["uid"].(float64)
+		if !ok {
+			return errors.New("invalid uid type for skill")
+		}
+
+		// Skip the base class
+		if uid == 0 {
+			continue
+		}
+
+		categoryUid, ok := rawSkill["category_uid"].(float64)
+		if !ok {
+			return fmt.Errorf("invalid category_uid type for skill %v", uid)
+		}
+
+		categoryName, ok := rawSkill["category_name"].(string)
+		if !ok {
+			return fmt.Errorf("invalid category_name type for skill %v", uid)
+		}
+
+		className, ok := rawSkill["caption"].(string)
+		if !ok {
+			return fmt.Errorf("invalid caption type for skill %v", uid)
+		}
+
+		skill := corev1alpha1.Skill{
+			ClassUid:     fmt.Sprintf("%v", uid),
+			CategoryUid:  fmt.Sprintf("%v", categoryUid),
+			CategoryName: &categoryName,
+			ClassName:    &className,
+		}
+
+		sv.mu.Lock()
+		sv.skills[skill.GetClassUid()] = &skill
+		sv.mu.Unlock()
+	}
+
+	return nil
+}
+
+func (sv *SkillValidator) HasSkill(classUid string) bool {
+	sv.mu.RLock()
+	defer sv.mu.RUnlock()
+	_, exists := sv.skills[classUid]
+
+	return exists
+}
+
+func (sv *SkillValidator) GetSkill(classUid string) *corev1alpha1.Skill {
+	sv.mu.RLock()
+	defer sv.mu.RUnlock()
+
+	return sv.skills[classUid]
+}
+
+func (sv *SkillValidator) GetSkillByName(className string) *corev1alpha1.Skill {
+	sv.mu.RLock()
+	defer sv.mu.RUnlock()
+	for _, skill := range sv.skills {
+		if skill.ClassName != nil && *skill.ClassName == className { //nolint:protogetter
+			return skill
+		}
+	}
+
+	return nil
+}
+
+func (sv *SkillValidator) Validate(skills []*corev1alpha1.Skill) error {
+	var errorMessages []string
+
+	for _, skill := range skills {
+		if skill.GetClassUid() == "" {
+			errorMessages = append(errorMessages, "skill class UID is required")
+			continue
+		}
+
+		if !sv.HasSkill(skill.GetClassUid()) {
+			errorMessages = append(errorMessages, skill.GetClassUid())
+		}
+
+		//TODO: Check if uid actually belongs to the provided class name, category uid and category name
+	}
+
+	if len(errorMessages) > 0 {
+		return fmt.Errorf("skills with uid %s not found", errorMessages)
```

Comment:
here you might find `"skill class UID is required"` in the uid slice

Reply:
Example error message now:

```
Error: failed to validate skills: validation failed with the following errors:
'class_uid' is required for skill category_uid:"1"  category_name:"Natural Language Processing"  class_name:"Text Completion".
'class_uid' not found for skill category_uid:"1"  class_uid:"107022"  category_name:"Natural Language Processing"  class_name:"Problem Solving".
```

Reply:
neat 👌